sirepo/sim_data/ml.py

```diff
@@ -62,3 +63,57 @@ def _lib_file_basenames(cls, data):
         if name:
             return [cls.lib_file_name_with_model_field("dataFile", "file", name)]
         return []
+
+
+class ArchiveManager:
+    import contextlib
+
+    _ARCHIVE_EXTENSIONS = (
+        ".tar.gz",
+        ".zip",
+    )
+
+    def __init__(self, file_path):
+        import zipfile
+        import tarfile
+
+        self.file_path = file_path
+        self.filename = file_path if isinstance(file_path, str) else file_path.basename
+        if self._is_archive_type(".zip"):
+            self.dir_check = "is_dir"
+            self.extractor = "open"
+            self.file_ctx = zipfile.ZipFile
+            self.item_name = "filename"
+            self.lister = "infolist"
+        if self._is_archive_type(".tar.gz"):
+            self.dir_check = "isdir"
+            self.extractor = "extractfile"
+            self.file_ctx = tarfile.open
+            self.item_name = "name"
+            self.lister = "getmembers"
+
```

Comment:
else? what happens if it is not a file that can be handled?

Reply:
Any file is supported, but if it's not a supported archive those properties are not used and the file is accessed directly

Reply:
[6e561ce](https://github.com/radiasoft/sirepo/pull/4736/commits/6e561ced430aef5c3176897462da7003d538df7e)